src/auction/src/priority_queue.mo

```diff
@@ -0,0 +1,42 @@
+import List "mo:base/List";
+import Order "mo:base/Order";
+
+module {
+
+  public type PriorityQueue<T> = List.List<T>;
+
+  public func next<T>(queue : PriorityQueue<T>) : ?T {
+    switch queue {
+      case null null;
+      case (?(h, _)) ?h;
+    };
+  };
+
+  // inserts item to the queue, places it after all items with higher or equal priority
+  public func insert<T>(queue : PriorityQueue<T>, item : T, comparePriority : (T, T) -> Order.Order) : PriorityQueue<T> {
```

Comment:
Should the compare function be passed every time to `insert`? Maybe we can make `type PriorityQueue<T>` a record and store it in there.

Reply:
I followed the code style of motoko-base modules such as list or tree. I think it works better because with this approach priority queue is a shareable type and don't need any share/unshare functionality.